src/BSON/Document.c

```diff
@@ -553,8 +554,14 @@ zval* php_phongo_document_read_property(phongo_compat_object_handler_type* objec
 	PHONGO_COMPAT_PROPERTY_ACCESSOR_NAME_TO_STRING(member, key, key_len);
 
 	if (!php_phongo_document_get(intern, key, key_len, rv)) {
-		// Exception already thrown
-		return &EG(uninitialized_zval);
+		if (type != BP_VAR_IS) {
+			// Exception already thrown
+			return &EG(uninitialized_zval);
+		}
+
+		// php_phongo_document_get will throw. If we're invoked with a BP_VAR_IS type, clear the exception and return a null value
+		zend_clear_exception();
```

Comment:
Is it worth reworking this to avoid throwing entirely? I imagine this would necessitate a change to `php_phongo_document_get()` (e.g. adding a `null_if_missing` bool), but the method is static to this file anyway so it'd be fairly self-contained.

This would also let you honor the more serious exception if `bson_iter_init()` fails, which should likely never happen in practice:

```
phongo_throw_exception(PHONGO_ERROR_RUNTIME, "Could not initialize BSON iterator");
```

Reply:
Added, makes sense to still throw if the BSON structure is invalid.